module/VuFind/src/VuFind/View/Helper/Root/Citation.php

```diff
@@ -192,6 +192,47 @@ public function getCitation($format)
         return '';
     }
 
+    protected function trimPunctuation($text)
```

Comment:
Can you use the existing stripPunctuation method?

Reply:
`stripPunctuation` was removing all punctuation when I just wanted to trim it from the ends.

- `stripPunctuation('J. J. J. Smith \.')` == 'J J J Smith'
- `trimPunctuation('J. J. J. Smith \.')` => 'J. J. J. Smith'

Reply:
Are you sure? The function comment is "Strip unwanted punctuation from the right side of a string." The code doesn't look to me like it should be doing anything other than trimming the end. If anything, it looks like it might not be aggressive enough for some scenarios.

Reply:
No, you're right. I misread the functionality. We'll keep only one at the end, regardless of which we go with.